packages/block-library/src/navigation/edit/index.js

```diff
@@ -154,7 +154,10 @@ function Navigation( {
 	} = useCreateNavigationMenu( clientId );
 
 	const createUntitledEmptyNavigationMenu = () => {
-		createNavigationMenu( '' );
+		return new Promise( async ( resolve ) => {
+			await createNavigationMenu( '' );
+			resolve();
+		} );
```

Comment:
Async functions will always return a `Promise` so you can remove the wrapper and just mark the function as async.